examples/src/main/scala/TracingExample.scala

```diff
@@ -21,24 +21,38 @@ import cats.effect.Resource
 import cats.effect.std.Console
 import cats.syntax.all._
 import io.opentelemetry.api.GlobalOpenTelemetry
+import org.typelevel.otel4s.Otel4s
+import org.typelevel.otel4s.TextMapPropagator
 import org.typelevel.otel4s.java.OtelJava
+import org.typelevel.otel4s.trace.Span
 import org.typelevel.otel4s.trace.Tracer
+import org.typelevel.vault.Vault
 
 import scala.concurrent.duration._
 
 trait Work[F[_]] {
-  def doWork: F[Unit]
+  def request(headers: Map[String, String]): F[Unit]
 }
 
 object Work {
-  def apply[F[_]: Monad: Tracer: Console]: Work[F] =
+  def apply[F[_]: Monad: Tracer: TextMapPropagator: Console]: Work[F] =
     new Work[F] {
-      def doWork: F[Unit] =
+      def request(headers: Map[String, String]): F[Unit] = {
         Tracer[F].span("Work.DoWork").use { span =>
-          span.addEvent("Starting the work.") *>
+          val vault = implicitly[TextMapPropagator[F]].extract(
+            Vault.empty,
```

Comment:
In Java, this would be `Context.current()`.

Conjecture: `Otel4s` should have a local `Vault`... which should be what we store the trace scope in!  This ties into #119.

Reply:
Makes sense.